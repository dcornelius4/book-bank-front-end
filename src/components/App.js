import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route
} from 'react-router-dom';
import LittleFreeLib from '../components/lfb/LittleFreeLib';
import Library from '../components/library/Library';
import BookClub from '../components/bookclub/BookClub';
import Home from './Home';
// import Layout from './layout/Layout';
import { NavigationBar } from '../components/layout/NavigationBar';
import { Jumbotron } from '../components/layout/Jumbotron';

export default function App() {
  return (
    <React.Fragment>
      <Router>
        <NavigationBar />
        <Jumbotron />
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/library" component={Library} />
          <Route path="/bookclub" component={BookClub} />
          <Route path="/littlefreelib" component={LittleFreeLib} />
        </Switch>
      </Router>
    </React.Fragment>
  );
}
