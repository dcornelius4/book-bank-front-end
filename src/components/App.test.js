import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

jest.mock('../services/auth.js', () => ({
  handleAuth() {
    return Promise.resolve([]);
  },
  handleCheck() {
    return Promise.resolve([]);
  }
}));

jest.mock('../services/libraryApi.js', () => ({
  makeLibrary() {
    return Promise.resolve([]);
  },
  getLibraries() {
    return Promise.resolve([]);
  },
  getLibrary() {
    return Promise.resolve([]);
  }
}));


describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
