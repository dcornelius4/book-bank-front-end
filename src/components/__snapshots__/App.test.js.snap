// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<Fragment>
  <BrowserRouter>
    <NavigationBar />
    <Jumbotron />
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(WithSession)",
            "type": [Function],
          }
        }
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Callback)",
            "type": [Function],
          }
        }
        path="/callback"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(WithSession)",
            "type": [Function],
          }
        }
        path="/library"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(WithSession)",
            "type": [Function],
          }
        }
        path="/add"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(WithSession)",
            "type": [Function],
          }
        }
        path="/littlefreelib"
      />
    </Switch>
  </BrowserRouter>
</Fragment>
`;
